Software/Development/SemesterProject/App/referenceHandler.c

```diff
@@ -19,16 +39,64 @@
  */
 static GUISignalsTag GUISignals;
 static float speedReference, torqueReference;
-static float deltaSpeed, deltaTorque;
+static float deltaSpeed = 5, deltaTorque = 0.1;
 static int undampedSpeedReference = 0;
+static int referenceSourceChanged = 0;
 
 
-static enum referenceSourceTag referenceSource = interfacePCB;
+static referenceSourceTag referenceSource = interfacePCB;
 static enum referenceTypeTag referenceType = torqueControl;
 
 
+void handleReferences(void)
+{
+    GUISignals = getGUISignals();
+    decideReferenceSource();
+    if (readSystemState() == RUNNING)
+    {
+        calculateReference();
+        decideReferenceType();
+    }
+}
+
+void decideReferenceSource(void)
```

Comment:
As the switch cases are the central part of the referenceHandler, i thinnk you should add comments here about the flow through it

Reply:
DOne